addon/-task-property.js

```diff
@@ -408,6 +408,10 @@ objectAssign(TaskProperty.prototype, propertyModifiers, {
   constructor: TaskProperty,
 
   setup(proto, taskName) {
+    if (this._hasSetMaxConcurrency && !this._hasSetBufferPolicy) {
```

Comment:
I thought about checking for Infinity but figured the warning wouldn't get logged if you manually set it to Infinity. The more I think about it... people probably aren't doing that. 

I'll make the change.